### PR TITLE
Update analytics to send to a specific SFRestAPI instance instead of the sharedInstance.

### PR DESCRIPTION
**Pull request by benrigas-sf**
This resolves a situation where logs can be sent to the wrong instance during a user switch.

### CODE REVIEW

File: libs/SalesforceSDKCore/SalesforceSDKCore/Classes/Analytics/SFSDKAnalyticsPublisher.h [hunk 1]
```diff
@@ -41,4 +43,13 @@ typedef void (^ _Nonnull PublishCompleteBlock)(BOOL success, NSError * _Nullable
...
 - (void) publish:(nonnull NSArray *) events publishCompleteBlock:(nonnull PublishCompleteBlock) publishCompleteBlock;
 
+/**
+* Publishes events to a network endpoint.
+*
+* @param events Events to be published.
+* @param user SFUserAccount to use to determine where events are to be published.
+* @param publishCompleteBlock Completion block invoked once network publish is complete.
+*/
+- (void) publish:(nonnull NSArray *) events user:(nullable SFUserAccount *)user publishCompleteBlock:(nonnull PublishCompleteBlock) publishCompleteBlock;
```
Comment (bhariharan) [inline libs/SalesforceSDKCore/SalesforceSDKCore/Classes/Analytics/SFSDKAnalyticsPublisher.h:53]: `SFUserAccount` should be marked `nonnull`.

File: libs/SalesforceSDKCore/SalesforceSDKCore/Classes/Analytics/SFSDKAnalyticsPublisher.h [hunk 2]
```diff
@@ -41,4 +43,13 @@ typedef void (^ _Nonnull PublishCompleteBlock)(BOOL success, NSError * _Nullable
  */
 - (void) publish:(nonnull NSArray *) events publishCompleteBlock:(nonnull PublishCompleteBlock) publishCompleteBlock;
```
Comment (bhariharan) [inline libs/SalesforceSDKCore/SalesforceSDKCore/Classes/Analytics/SFSDKAnalyticsPublisher.h:44]: Mark this method deprecated in favor of the one that you have added below. See an example on how to do this [here](https://github.com/forcedotcom/SalesforceMobileSDK-iOS/blob/dev/libs/SalesforceSDKCore/SalesforceSDKCore/Classes/Security/SFPasscodeKeyStore.h#L34).

File: libs/SalesforceSDKCore/SalesforceSDKCore/Classes/Analytics/SFSDKAILTNPublisher.m [hunk 3]
```diff
@@ -65,12 +71,13 @@ + (void) publishLogLines:(NSDictionary *) bodyDictionary publishCompleteBlock:(P
     [request setCustomRequestBodyData:postData contentType:@"application/json"];
     [request setHeaderValue:@"gzip" forHeaderName:@"Content-Encoding"];
     [request setHeaderValue:[NSString stringWithFormat:@"%lu", (unsigned long)[postData length]] forHeaderName:@"Content-Length"];
-    [[SFRestAPI sharedInstance] sendRequest:request failureBlock:^(id response, NSError *e, NSURLResponse *rawResponse) {
+
+    [restAPI sendRequest:request failureBlock:^(id  _Nullable response, NSError * _Nullable e, NSURLResponse * _Nullable rawResponse) {
```
Comment (bhariharan) [inline libs/SalesforceSDKCore/SalesforceSDKCore/Classes/Analytics/SFSDKAILTNPublisher.m:75]: Why have nullability markers been added here? They should be added in the definition already, I don't think we need them here.

**Review by bhariharan — CHANGES_REQUESTED** [review]

**Comment by bhariharan** [comment]
Could you also sign the CLA, @benrigas-sf?

File: libs/SalesforceSDKCore/SalesforceSDKCore/Classes/Analytics/SFSDKAILTNPublisher.m (hunk 3 above)
Comment (benrigas-sf) [reply libs/SalesforceSDKCore/SalesforceSDKCore/Classes/Analytics/SFSDKAILTNPublisher.m:75]: Those got added automatically with code completion, I'll remove them no problem.

File: libs/SalesforceSDKCore/SalesforceSDKCore/Classes/Analytics/SFSDKAnalyticsPublisher.h (hunk 2 above)
Comment (benrigas-sf) [reply libs/SalesforceSDKCore/SalesforceSDKCore/Classes/Analytics/SFSDKAnalyticsPublisher.h:44]: Will do

**Comment by benrigas-sf** [comment]
@bhariharan I filled out the google form and got added to the org, do I still need to sign it?

**Review by bhariharan — APPROVED** [review]

**Pull request merged**